rcl_logging_spdlog/test/test_logging_interface.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rcutils/allocator.h>
+#include <rcutils/error_handling.h>
+#include "rcl_logging_spdlog/logging_interface.h"
+#include "gtest/gtest.h"
+
+static void * bad_malloc(size_t, void *)
+{
+  return nullptr;
+}
+
+TEST(logging, init_invalid)
+{
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rcutils_allocator_t bad_allocator = rcutils_get_default_allocator();
+  bad_allocator.allocate = bad_malloc;
+
+  // Config files are not supported by spdlog
+  ASSERT_NE(0, rcl_logging_external_initialize("anything", allocator));
```

Comment:
Use explicit error level
```
ASSERT_EQ(RCL_LOGGING_RET_ERROR, ...
```

Reply:
I can't do that. Those macros are defined in the C++ file, not the header.

The documentation in `rcl` for these functions actually says that the return values should be `RCL_RET_OK` or `RCL_RET_ERROR`, so this implementation doesn't match that.

I doubt that anyone has taken a dependency on the error codes for these functions, but we're past API freeze for Foxy. Thoughts? Fix it (and log4cxx) in G-turtule?

Reply:
My primary suggestion was to use ASSERT_EQ and the actual error code that is returned, so I guess the suggestion with your other point would be: 
```
ASSERT_EQ(2, rcl_logging_external_initialize...)
```
If you can't use the actual macro, that's not a big deal to me.

As for whether to change the return codes, you might want to get input about the API fix from a more seasoned maintainer, but I don't see an issue with fixing something to conform to the actually expected API of the larger ecosystem. This package has yet to be bumped to 1.0, so you might be able to squeeze it in.

Reply:
I'm not super jazzed about specifically asserting undocumented behavior in the tests, but you seem confident that it's the right thing to do, so I'll roll with it. Updated in 7d52599.

Reply:
Yah, the undocumented behavior isn't great, and you might want to create a follow-up issue. That being said, I like to view tests as another sort of "documentation". If somebody wants to fix the API in the future, the test doesn't silently acquiesce to a new API. It will have to be changed as well, which also means that it will resist unwanted changes in the future. Meanwhile, people can look at the test and see the warts and all of the current API.

I also love being as explicit as possible that the expected error matches the actual error as much as possible. I would add EXPECT_STREQ with rcutil's error message if it also didn't report the file path in that message.